### DISCUSSION TITLE
Tips on how to debug Install-ChocolateyInstallPackage?

### DISCUSSION THREAD

**Discussion by basictheprogram** [chocolatey-community/chocolatey-packages#2169] [General]
Building, testing, and deploying a chocolatey package via gitlab CD/CI using Virtualbox 7.0 and Windows10 guest.
```powershell
$ErrorActionPreference = 'Stop';
$data = & (Join-Path -Path (Split-Path -Path $MyInvocation.MyCommand.Path) -ChildPath data.ps1)
$toolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
$setupName = 'SetupDmD_R2_9_4_Production.exe'

$packageArgs = @{
    packageName    = $env:ChocolateyPackageName
    unzipLocation  = $toolsDir
    fileType       = 'EXE'
    silentArgs     = '/s /f1".\setup.iss"'
    softwareName   = 'Do-more Designer'
    validExitCodes = @(0, 3010, 1641)
    url                        = $data.url
    checksum           = $data.checksum
    checksumType   = $data.checksumType
}
Install-ChocolateyZipPackage @packageArgs
$packageArgs.file = Join-Path -Path $toolsDir -ChildPath $setupName
Install-ChocolateyInstallPackage @packageArgs
```
CI/CD log snippet
```
C:\ProgramData\chocolatey\lib\do-more\tools
403Running Install-ChocolateyInstallPackage -packageName 'do-more' -fileType 'EXE' -validExitCodes '0 3010 1641' -silentArgs '/s /f1".\setup.iss"' -file 'C:\ProgramData\chocolatey\lib\do-more\tools\SetupDmD_R2_9_4_Production.exe' 
404Running Get-OSArchitectureWidth -compare '32' 
405Installing do-more...
406Running Start-ChocolateyProcessAsAdmin -validExitCodes '0 3010 1641' -workingDirectory 'C:\ProgramData\chocolatey\lib\do-more\tools' -statements '/s /f1".\setup.iss" ' -exeToRun 'C:\ProgramData\chocolatey\lib\do-more\tools\SetupDmD_R2_9_4_Production.exe' 
407Test-ProcessAdminRights: returning True
408Elevating permissions and running ["C:\ProgramData\chocolatey\lib\do-more\tools\SetupDmD_R2_9_4_Production.exe" /s /f1".\setup.iss" ]. This may take a while, depending on the statements.
409Command ["C:\ProgramData\chocolatey\lib\do-more\tools\SetupDmD_R2_9_4_Production.exe" /s /f1".\setup.iss" ] exited with '-2147213312'.
410ERROR: Running ["C:\ProgramData\chocolatey\lib\do-more\tools\SetupDmD_R2_9_4_Production.exe" /s /f1".\setup.iss" ] was not successful. Exit code was '-2147213312'. See log for possible error messages.
```
Google says `Exit code was '-2147213312'` is potentially due to a corrupted or missing  .iss file. 
I am confident the .iss file is in the `C:\ProgramData\chocolatey\lib\do-more\tools` directory and not corrupt (see below).

Trying manual(?) debugging(?) by going into the lib-bad directory and simulating the CI/CD chocolatey commands.
```
> cd C:\ProgramData\chocolatey\lib-bad\do-more\tools
> dir
    Directory: C:\ProgramData\chocolatey\lib-bad\do-more\tools

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---            3/8/2023 10:11 AM          19756 .chocolateyInstall.ps1.un~
-a---            3/8/2023 10:11 AM            978 chocolateyInstall.ps1
-a---            3/8/2023 10:11 AM           1195 chocolateyInstall.ps1~
-a---            3/8/2023 10:11 AM            198 data.ps1
-a---            3/8/2023 11:31 AM           1204 setup.iss
-a---            3/9/2023 10:58 AM            189 setup.log
-a---           6/20/2022  1:21 PM      310341384 SetupDmD_R2_9_4_Production.exe
-a---            3/8/2023 10:14 AM              6 SetupDmD_R2_9_4_Production.exe.ignore
-a---           6/20/2022  1:19 PM        6221215 Updates.pdf
> C:\ProgramData\chocolatey\lib-bad\do-more\tools\SetupDmD_R2_9_4_Production.exe /s /f1".\setup.iss"
> $LASTEXITCODE
0
```
Any advise on how to debug this problem further?

Thanks.

**Comment by pauby** [comment]
@basictheprogram I'm not sure I understand why this was posted here as it doesn't appear to be related to the repository?

**Reply by basictheprogram** [reply]
I posted similar question, https://github.com/chocolatey-community/chocolatey-packages/discussions/2154, to General and you helped me on that question. Should I ask my chocolatey package/chocolatey maintainer questions somewhere else?

**Reply by pauby** [reply]
The [answer](https://github.com/orgs/chocolatey-community/discussions/2154#discussioncomment-5166181) in #2154 asked for questions for packages not in this repository to be posted to [Discord](https://ch0.co/community).

This repository is for Chocolatey Community Chocolatey Packages. Any questions should be based on packages in the repository.

**Reply by basictheprogram** [reply]
Build the package by hand. Submitted it.

Package Url: https://community.chocolatey.org/packages/do-more/2.9.4
Maintainer(s): basictheprogram

Would prefer to build the package with gitlab CI/CD. 

Maybe like my last question CI/CD cannot be used to build this package. Which would be disappointing.

**Reply by pauby** [reply]
> This repository is for Chocolatey Community Chocolatey Packages. Any questions should be based on packages in **this GitHub** repository.

Added emphasis.

This is not the [Chocolatey Community Repository](https://community.chocolatey.org). This is a repository that contains packages that the Chocolatey Community maintain.

**Reply by basictheprogram** [reply]
If you do not want to help. Just say you do not want to help. I get it. Please direct me where I can ask the question.

The issue https://github.com/orgs/chocolatey-community/discussions/2154 was also a package not in the community package feed, but you helped and gave advise on that package.

I am trying to get the `Do-more` package into the community feed. Just like I was trying to get the `statistiCA` into the community feed. Questions are similar. Given your responses. I guess the questions are not similar.

I will go look for a more appropriate venue to ask my question.

**Reply by pauby** [reply]
> If you do not want to help. Just say you do not want to help. I get it

We pointed you to the best place to get help, above and in #2154.

> Please direct me where I can ask the question.

Please see the link for Discord above.

**Comment by AdmiringWorm** [comment]
As the discussion here has run its course, it isn't relevant for this repository, and an answer has been provided on where you can get help. I will be closing and locking this issue now.
